packages/core/admin/admin/src/content-manager/components/DynamicTable/CellContent/utils/hasContent.js

```diff
@@ -0,0 +1,19 @@
+export default function hasContent(type, content, metadatas, fieldSchema) {
+  let normalizedContent = content;
+
+  if (type === 'component') {
+    const { mainField } = metadatas;
+
+    if (fieldSchema.repeatable) {
+      normalizedContent = content?.[0]?.[mainField];
```

Comment:
This is not a truthy statement as the value at the 0 index can be not empty but the mainField could.

```js
return content.every(value => !isEmpty(value)
```